File: trueface-sdk-android/sdk/src/main/java/ai/trueface/sdk/models/MatchData.java
package ai.trueface.sdk.models;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class MatchData {

    @SerializedName("emb2_match")
    @Expose
    private boolean emb2Match;

    @SerializedName("emb0_score")
    @Expose
    private double emb0Score;

    @SerializedName("emb2_score")
    @Expose
    private double emb2Score;

    @SerializedName("emb1_score")
    @Expose
    private double emb1Score;

    @SerializedName("emb1_match")
    @Expose
    private boolean emb1Match;

    @SerializedName("emb0_match")
    @Expose
    private boolean emb0Match;

    public boolean isEmb2Match() {
        return emb2Match;
    }

    public void setEmb2Match(boolean emb2Match) {
        this.emb2Match = emb2Match;
    }

    public double getEmb0Score() {
        return emb0Score;
    }

    public void setEmb0Score(double emb0Score) {
        this.emb0Score = emb0Score;
    }

    public double getEmb2Score() {
        return emb2Score;
    }

    public void setEmb2Score(double emb2Score) {
        this.emb2Score = emb2Score;
    }

    public double getEmb1Score() {
        return emb1Score;
    }

    public void setEmb1Score(double emb1Score) {
        this.emb1Score = emb1Score;
    }

    public boolean isEmb1Match() {
        return emb1Match;
    }

    public void setEmb1Match(boolean emb1Match) {
        this.emb1Match = emb1Match;
    }

    public boolean isEmb0Match() {
        return emb0Match;
    }

    public void setEmb0Match(boolean emb0Match) {
        this.emb0Match = emb0Match;
    }

}

File: trueface-sdk-android/sdk/src/main/java/ai/trueface/sdk/models/MatchData.kt
package ai.trueface.sdk.models

import android.os.Parcel
import android.os.Parcelable
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

class MatchData() : Parcelable {

    @SerializedName("emb2_match")
    @Expose
    var emb2Match: Boolean? = null

    @SerializedName("emb0_score")
    @Expose
    var emb0Score: Double? = null

    @SerializedName("emb2_score")
    @Expose
    var emb2Score: Double? = null

    @SerializedName("emb1_score")
    @Expose
    var emb1Score: Double? = null

    @SerializedName("emb1_match")
    @Expose
    var emb1Match: Boolean? = null

    @SerializedName("emb0_match")
    @Expose
    var emb0Match: Boolean? = null

    constructor(parcel: Parcel) : this() {
        emb2Match = parcel.readValue(Boolean::class.java.classLoader) as? Boolean
        emb0Score = parcel.readValue(Double::class.java.classLoader) as? Double
        emb2Score = parcel.readValue(Double::class.java.classLoader) as? Double
        emb1Score = parcel.readValue(Double::class.java.classLoader) as? Double
        emb1Match = parcel.readValue(Boolean::class.java.classLoader) as? Boolean
        emb0Match = parcel.readValue(Boolean::class.java.classLoader) as? Boolean
    }

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeValue(emb2Match)
        parcel.writeValue(emb0Score)
        parcel.writeValue(emb2Score)
        parcel.writeValue(emb1Score)
        parcel.writeValue(emb1Match)
        parcel.writeValue(emb0Match)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<MatchData> {
        override fun createFromParcel(parcel: Parcel): MatchData {
            return MatchData(parcel)
        }

        override fun newArray(size: Int): Array<MatchData?> {
            return arrayOfNulls(size)
        }
    }

}